### PR TITLE
Adding metrics to the ASG

### PR DESCRIPTION
**Pull request by triwats**
##### Corresponding Issue(s):

* Unable to add issues within repository

##### Summary of change(s):

* Add definition for metric parameter map to allow for ASG Cloudwatch metrics to be available in tf-v11 version of this module as it currently is with v12 and newer outside of the scope of CW alarms.

##### Reason for Change(s):

There are currently no metrics for an ASG. In order to enable metrics in the Terraform v11 version of this using this module, we need to go in and manually enable it. Each apply, this is disabled again. This makes it problematic when you're trying to monitor and view scaling.

##### Will the change trigger resource destruction or replacement? If yes, please provide justification:

No, just an alter in place.

##### Does this update/change involve issues with other external modules? If so, please describe the scenario.

Negative

##### If input variables or output variables have changed or has been added, have you updated the README?

Yes

##### Do examples need to be updated based on changes?

It might be a good idea, but if required then please advise and I can put them in there. 

I am still testing this on a forked project with our code base, but thought I would open this first to have potential discussions around this.

### CODE REVIEW

**Comment by gdelvalle** [comment]
This is a request to backport changes similar to #63.

**Comment by triwats** [comment]
Removing draft. I've formatted this too using our projects version as provided by the RS team: 

~/src/tf-raws-uat master
❯ tf --version
Terraform v0.11.14

File: README.md [hunk 1]
```diff
@@ -62,6 +62,9 @@ Using [aws-terraform-cloudwatch\_alarm](https://github.com/rackspace-infrastruct
...
 | enable\_scaling\_actions | Should this autoscaling group be configured with scaling alarms to manage the desired count.  Set this variable to false if another process will manage the desired count, such as EKS Cluster Autoscaler. | `string` | `true` | no |
 | enable\_scaling\_notification | true or false. If 'scaling\_notification\_topic' is set to a non-empty string, this must be set to true. Otherwise, set to false. This variable exists due to a terraform limitation with using count and computed values as conditionals | `string` | `false` | no |
+| enabled\_asg\_metrics | List of ASG metrics desired.  This can only contain the following values: `GroupDesiredCapacity`, `GroupInServiceCapacity`, `GroupPendingCapacity`, `GroupMinSize`, `GroupMaxSize`, `GroupInServiceInstances`, `GroupPendingInstances`, `GroupStandbyInstances`, `GroupStandbyCapacity`, `GroupTerminatingCapacity`, `GroupTerminatingInstances`, `GroupTotalCapacity`, `GroupTotalInstances`. | `list(string)` | `[]` | no |
+| encrypt\_primary\_ebs\_volume | Encrypt root EBS Volume? true or false | `bool` | `false` | no |
+| encrypt\_primary\_ebs\_volume | Encrypt root EBS Volume? t
```
Comment (gdelvalle) [inline README.md:67]: You've got duplicate metrics  defined in the readme `encrypt\_primary\_ebs\_volume`

**Review by gdelvalle — CHANGES_REQUESTED** [review]
- we have a duplicate metric in the readme
- Similar to #63 , could you fix the alarm name in group_terminating_instances module.  The Last  `}`  is superfluous.  It was a lingering error for a while.
```
module "group_terminating_instances" {
  source = "git@github.com:rackspace-infrastructure-automation/aws-terraform-cloudwatch_alarm//?ref=v0.0.1"
  alarm_count              = "${var.asg_count}"
  alarm_description        = "Over ${var.terminated_instances} instances terminated in last 6 hours, generating ticket to investigate."
  alarm_name               = "${var.resource_name}-GroupTerminatingInstances}"
  comparison_operator      = "GreaterThanThreshold"
  dimensions               = "${data.null_data_source.alarm_dimensions.*.outputs}"
  evaluation_periods       = 1
  metric_name              = "GroupTerminatingInstances"
  namespace                = "AWS/AutoScaling"
  notification_topic       = "${var.notification_topic}"
```
- I think  for the tests to even work, you'll need to remove the scaleFT install option.  This was done for the 0.12 branch in #62

File: README.md (hunk 1 above)
Comment (triwats) [reply README.md:67]: Got a bit fat fingered in vim, fixed

**Comment by triwats** [comment]
Hey again @gdelvalle - thanks for the initial review and valid feedback!

**1. We have a duplicate metric in the readme**

I've fixed this

**2. Similar to #63 , could you fix the alarm name in group_terminating_instances module. The Last } is superfluous. It was a lingering error for a while.**

Noted, this has now been resolved and should evaluate correctly

**3. I think for the tests to even work, you'll need to remove the scaleFT install option. This was done for the 0.12 branch in #62**

I'm not really sure how to proceed with this one as I'm not too sure what ScaleFT does, I can try to put something in but having a bit of an understanding could help, as the blocks of definition in the test around ScaleFT are seemingly larger in the v11 version of this module.

**Comment by gdelvalle** [comment]
Hi @triwats,
> I'm not really sure how to proceed with this one as I'm not too sure what ScaleFT does, I can try to put something in but having a bit of an understanding could help, as the blocks of definition in the test around ScaleFT are seemingly larger in the v11 version of this module.

Sorry about being vague here.  ScaleFT was a Rackspace specific on instance agent we moved away from.  Your changes  seem good and knocking out the scaleFT stuff isn't necessary on your end - I'll get to that.  

One problem that isn't your fault,  automated tests don't  run from forks.  I'll have to copy your changes over and I'll add  in the  scaleFT removal then.

**Comment by triwats** [comment]
Hey again @gdelvalle, thanks for the explanation around ScaleFT.

I've just pushed another commit which removes any references to this that I could find and makes a note in the README. This needs another review but hpefully it makes it easier for you to just copy from the fork directly and test it. 

Let me know if you would like to make any more changes!

**Comment by gdelvalle** [comment]
Hi @triwats ,  thanks for you work.  Per my earlier  comment, to get automated tests to run, I've ported your changes  over to #67  and will close this PR. Please  follow  in that thread for updates and release information.  Thanks again!

**Comment by gdelvalle** [comment]
released here: https://github.com/rackspace-infrastructure-automation/aws-terraform-ec2_asg/releases/tag/v0.0.27

**Comment by triwats** [comment]
Great stuff @gdelvalle - thanks for all the help with this!

**Pull request closed without merging**